src/EnergyPlus/OutputReports.cc

```diff
@@ -1194,22 +1195,24 @@ void DetailsForSurfaces(EnergyPlusData &state, int const RptType) // (1=Vertices
                            << format("{:.2R}", thisSurface.Tilt) << "," << format("{:.2R}", thisSurface.Width) << ","
                            << format("{:.2R}", thisSurface.Height) << "," << format("{:.2R}", thisSurface.Reveal) << ",";
 
-                static constexpr std::array<std::string_view, (int)ConvectionConstants::ConvCoefOverrideType::Num> overrideTypeStrs = {
+                static constexpr std::array<std::string_view, (int)Convect::OverrideType::Num> overrideTypeStrs = {
                     "User Supplied Value", "User Supplied Schedule", "User Supplied Curve", "User Specified Model"};
 
-                if (state.dataSurface->SurfIntConvCoeffIndex(surf) > 0) {
```

Comment:
Using `SurfIntConvCoeffIndex` to refer to both built-in coefficients (negative numbers) and user-defined coefficients (positive numbers) was a mess. There are now two variables: `SurfIntConvCoeff` which is an enumeration (with `Value`, `Schedule`, and `Curve` as possible values), and `SurfIntConvUserCoeffNum` which is an index to the user-defined coefficient struct when `SurfIntConvCoeff` is `Value`, `Schedule`, or `Curve`.